frame/contracts/src/exec.rs

```diff
@@ -1328,14 +1328,16 @@ where
 		&mut self.top_frame_mut().nested_gas
 	}
 
-	fn append_debug_buffer(&mut self, msg: &str) -> bool {
+	fn append_debug_buffer(&mut self, msg: &str) -> Result<bool, DispatchError> {
 		if let Some(buffer) = &mut self.debug_message {
 			if !msg.is_empty() {
-				buffer.extend(msg.as_bytes());
+				buffer.try_append(&mut msg.as_bytes().to_vec()).map_err(|_| {
+					DispatchError::Other("Maximum allowed debug buffer size exhausted!")
```

Comment:
Please don't add `Other` errors. The message is dropped. Create a new module error.